spring-integration-core/src/test/java/org/springframework/integration/message/ExpressionEvaluatingMessageHandlerTests.java

```diff
@@ -76,20 +77,23 @@ public void validExpressionWithSomeArgs() {
 		handler.handleMessage(new GenericMessage<String>("testtest", headers));
 	}
 
-	@Test(expected = MessagingException.class)
+	@Test
 	public void expressionWithReturnValue() {
 		Message<?> message = new GenericMessage<Float>(.1f);
-		try {
-			Expression expression = parser.parseExpression("T(System).out.printf('$%4.2f', payload)");
-			ExpressionEvaluatingMessageHandler handler = new ExpressionEvaluatingMessageHandler(expression);
-			handler.setBeanFactory(mock(BeanFactory.class));
-			handler.afterPropertiesSet();
-			handler.handleMessage(message);
-		}
-		catch (MessagingException e) {
-			assertThat(message).isEqualTo(e.getFailedMessage());
-			throw e;
-		}
+		Expression expression = parser.parseExpression("T(System).out.printf('$%4.2f', payload)");
+		ExpressionEvaluatingMessageHandler handler = new ExpressionEvaluatingMessageHandler(expression);
+		handler.setBeanFactory(mock(BeanFactory.class));
+		handler.afterPropertiesSet();
+		assertThatThrownBy(() -> {
+			try {
+				handler.handleMessage(message);
+			}
+			catch (MessagingException e) {
+				assertThat(message).isEqualTo(e.getFailedMessage());
+				throw e;
+			}
+		}).isInstanceOf(MessagingException.class);
```

Comment:
There  is an AssertJ API to extract object properties smooth way without this composition which maybe lost.
Something like:
```
		assertThatExceptionOfType(MessagingException.class)
				.isThrownBy(() -> handler.handleMessage(message))
				.extracting(MessagingException::getFailedMessage)
				.isEqualTo(message);
```